sktime/transformations/series/adi_cv.py

```diff
@@ -153,37 +174,49 @@ def _transform(self, X, y=None):
             during initialization. Specifically, the columns include (by default):
 
                 1. Average Demand Interval (ADI)
-                2. Variance (CV2)
+                2. Coefficient of Variation squared (CV2)
                 3. categorical class
         """
-        X_non_zero = X.to_numpy().nonzero()
-        X_non_zero = X.iloc[X_non_zero]
+        X_non_zero_indices = X.to_numpy().nonzero()[0]
+        X_non_zero_values = X.iloc[X_non_zero_indices].values
 
-        # Calculating ADI value based on formula from paper
-        adi_value = len(X) / (len(X_non_zero) - 1)
+        # Calculating ADI value based on adi_trim_handling parameter
+        n_non_zero_observations = len(X_non_zero_indices)
+        n_observations = len(X)
 
-        # Calculating variance for all non-zero values
-        variance = X_non_zero.var().iloc[0]
-        cv2_value = variance / len(X_non_zero)
+        if self.adi_trim_handling == "trim":
+            first_non_zero = X_non_zero_indices[0]
+            last_non_zero = X_non_zero_indices[-1]
 
-        # Calculating the class type
+            adi_numerator = last_non_zero - first_non_zero
+            adi_denominator = n_non_zero_observations - 1
 
-        adi_low = adi_value <= self.adi_threshold
```

Comment:
I would leave the old logic of computing `cv2_low` first - that makes the code easier to read, and debug. May I kindly request to add back `adi_high`, `adi_low`, etc?